samples/interactive-tutorials/setup/setup_cleanup.js

```diff
@@ -0,0 +1,338 @@
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+'use strict';
+// Imports the Google Cloud client library.
+const {ProductServiceClient} = require('@google-cloud/retail').v2;
+const {UserEventServiceClient} = require('@google-cloud/retail').v2;
+const {Storage} = require('@google-cloud/storage');
+const {BigQuery} = require('@google-cloud/bigquery');
+const fs = require('fs');
+
+const delay = ms => new Promise(resolve => setTimeout(resolve, ms));
+
+const createProduct = async (
+  projectNumber,
+  generatedProductId,
+  isFullfillment = false
+) => {
+  // The parent catalog resource name
+  const parent = `projects/${projectNumber}/locations/global/catalogs/default_catalog/branches/default_branch`;
+  const apiEndpoint = 'retail.googleapis.com';
+
+  // The ID to use for the product
+  const productId = generatedProductId
+    ? generatedProductId
+    : Math.random().toString(36).slice(2).toUpperCase();
+
+  const fulfillmentInfo = isFullfillment
+    ? [
+        {
+          type: 'same-day-delivery',
+          placeIds: ['store1', 'store2', 'store3'],
+        },
+      ]
+    : [];
+
+  // The product to create.
+  const product = {
+    title: 'Nest Mini',
+    type: 'PRIMARY',
+    categories: ['Speakers and displays'],
+    brands: ['Google'],
+    fulfillmentInfo,
+    priceInfo: {
+      price: 30.0,
+      originalPrice: 35.5,
+      currency_code: 'USD',
+    },
+    availability: 'IN_STOCK',
+  };
+
+  const retailClient = new ProductServiceClient({apiEndpoint});
+
+  // Construct request
+  const request = {
+    parent,
+    product,
+    productId,
+  };
+
+  // Run request
+  const response = await retailClient.createProduct(request);
+  console.log(`Product ${response[0].id} created`);
+  return response[0];
+};
+
+const getProduct = async name => {
+  const apiEndpoint = 'retail.googleapis.com';
+  const retailClient = new ProductServiceClient({apiEndpoint});
+
+  // Construct request
+  const request = {
+    name,
+  };
+
+  // Run request
+  const response = await retailClient.getProduct(request);
+  return response;
+};
+
+const deleteProduct = async name => {
+  const apiEndpoint = 'retail.googleapis.com';
+  const retailClient = new ProductServiceClient({apiEndpoint});
+
+  // Construct request
+  const request = {
+    name,
+  };
+
+  // Run request
+  const response = await retailClient.deleteProduct(request);
+  return response;
+};
+
+const deleteProducts = async (projectNumber, ids) => {
+  const apiEndpoint = 'retail.googleapis.com';
+  const retailClient = new ProductServiceClient({apiEndpoint});
+
+  for (let i = 0; i < ids.length; ++i) {
```

Comment:
There's a risk that this will break when two sets of tests are running in parallel I think?

What I would perhaps do instead, would be to delete any products older than a certain number of hours (is there a creation time on the product?).

Reply:
We have no risk there because we pass inside this method products IDs generated during a test. Each test generates unique IDs for products. It's impossible to remove the same ID from different tests.

Reply:
If our test fails periodically (imagine a connectivity issue to the API during the deletion of the first ID), will we gradually leak products?

I would recommend adding a helper that deletes all product IDs, but only if they're older than a certain date, then run this once when the test suite is starting up. Then you'll never need to come back and cleanup leaked products.

---

If you open a tracking issue for this work, I'd be okay with it being in a separate PR.

Reply:
Actually we don't run the cleanup after each test run. We are using the long-living test resources.
The was a request from Karl to create the scripts for creating and removing the test resources, which will be run ocasionally  if the project resources such as products in a catalog of bigQuery tables of GCS objects are lost. So this PR is providing these scripts.

In the RemoveTestResources there is a delete_all_products method